SuperBuild/External_ffi.cmake

```diff
@@ -0,0 +1,86 @@
+
+set(proj ffi)
+
+# Set dependency list
+set(${proj}_DEPENDENCIES "")
+
+# Include dependent projects if any
+ExternalProject_Include_Dependencies(${proj} PROJECT_VAR proj DEPENDS_VAR ${proj}_DEPENDENCIES)
+
+if(Slicer_USE_SYSTEM_${proj})
+  unset(ffi_DIR CACHE)
+  find_package(FFI REQUIRED)
+  set(FFI_INCLUDE_DIR ${FFI_INCLUDE_DIRS})
+  set(FFI_LIBRARY ${FFI_LIBRARIES})
+endif()
+
+# Sanity checks
+if(DEFINED ffi_DIR AND NOT EXISTS ${ffi_DIR})
+  message(FATAL_ERROR "ffi_DIR variable is defined but corresponds to nonexistent directory")
+endif()
+
+if(NOT DEFINED ffi_DIR AND NOT Slicer_USE_SYSTEM_${proj})
+
+  set(EP_SOURCE_DIR ${CMAKE_BINARY_DIR}/${proj})
+  set(EP_BINARY_DIR ${CMAKE_BINARY_DIR}/${proj}-build)
+  set(EP_INSTALL_DIR ${CMAKE_BINARY_DIR}/${proj}-install)
+
+  ExternalProject_SetIfNotDefined(
+    Slicer_${proj}_GIT_REPOSITORY
+    "${EP_GIT_PROTOCOL}://github.com/fbordignon/libffi.git"
```

Comment:
Before integrating, we will create a fork of `libffi/libffi` into the Slicer organization that will:
* have a default `welcome` branch describing the fork and available branch
* history of attempt to contribute CMake support to libffi
* describe the naming convention for branches

It will be similar to what we have for these projects:
https://github.com/search?q=topic%3A3d-slicer-fork+org%3ASlicer+fork%3Atrue